.github/workflows/tests.yml

```diff
@@ -1,25 +1,36 @@
-name: continuous-integration
+name: tests
 
 on: [push, pull_request]
 
 jobs:
 
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v3
+      with:
+        python-version: 3.8
+    - uses: pre-commit/action@v2.0.0
+
   tests:
 
     strategy:
       matrix:
-        python-version: [3.7, 3.8]
+        python-version: ["3.8", "3.11", " 3.12"]
```

Comment:
any reason why you are not testing all python version on ubuntu and only latest on extra platform (mac and windows). This is what I have been doing so far in other repo but now I'm questioning myself....